scheduler-server/scheduling-dsl-compiler/src/libs/scheduler-edsl-fluent-api.ts

```diff
@@ -46,33 +46,72 @@ export class Goal {
       interval: opts.interval,
     });
   }
-  public static CoexistenceGoal(opts: { activityTemplate: ActivityTemplate, forEach: ActivityType }): ActivityCoexistenceGoal {
+  public static CoexistenceGoal(opts: { activityTemplate: ActivityTemplate, forEach: WindowSet }): ActivityCoexistenceGoal {
     return Goal.new({
       kind: AST.NodeKind.ActivityCoexistenceGoal,
       activityTemplate: opts.activityTemplate,
-      forEach: {
-        type: opts.forEach
-      },
+      forEach: opts.forEach.__serialize(),
     });
   }
 }
 
+class WindowSet {
+  protected readonly windowSpecifier: AST.WindowsExpression;
+  private constructor(windowSpecifer: AST.WindowsExpression) {
+    this.windowSpecifier = windowSpecifer;
+  }
+  private static new(windowSpecifier: AST.WindowsExpression) {
+    return new WindowSet(windowSpecifier);
+  }
+  public __serialize() {
+    return this.windowSpecifier;
+  }
+
+  public static gt(resource: Resource, value: Double): WindowSet {
+    return WindowSet.new({
+      kind: 'WindowsExpressionGreaterThan',
```

Comment:
That's a good idea - playing around with it now, it seems like intellisense works better with the enum than it did with the union type, probably because of the additional context:

<img width="798" alt="Screen Shot 2022-05-04 at 12 40 01 PM" src="https://user-images.githubusercontent.com/1189602/166813205-216b84f6-473e-45f1-8792-2ac44e5b46a6.png">

Reply:
Might make sense to have a super intuitive mapping to a name that doesn't require the brackets and quotes - I think its not particularly intuitive to have to do the string indexing. Also not sure if that is really a viable option the way it was for command stems.

Reply:
I would think the most intuitive form of a resource name is the exact string (any mangling we do would make it less intuitive in my opinion... but maybe that's a Basak/users question).

Reply:
I agree that the most intuitive form of the name is the exact string, but that requires a less intuitive string indexing. I think the question is which is the lesser evil.